src/util/NuulsUploader.hpp

```diff
@@ -0,0 +1,18 @@
+#include "common/Channel.hpp"
+#include "widgets/helper/ResizingTextEdit.hpp"
+
+#include <QMimeData>
+#include <QString>
+
+namespace chatterino {
+struct RawImageData {
+    QByteArray data;
+    QString format;
+};
+void upload(QByteArray imageData, ChannelPtr channel,
+            ResizingTextEdit &textEdit, std::string format);
+void upload(RawImageData imageData, ChannelPtr channel,
+            ResizingTextEdit &textEdit);
+void upload(const QMimeData *source, ChannelPtr channel,
+            ResizingTextEdit &outputTextEdit);
```

Comment:
Didn't know you could highlight more, this is what should have a `public` before it.

Reply:
Well, these functions are not part of a class anyway -- visibility modifiers don't make sense here.

Reply:
NuulsUploader isn't a singleton and from what I found everything in a namespace is public.

Reply:
you're correct, I thought it was a class